File: dehaze-react-native/src/pages/dataset/services/datasetApi.ts
/**
 * 数据集模块 API 封装
 *
 * 直接调用 SDK 的 DatasetAPI / DatasetItemAPI / ItemFileAPI，
 * SDK 内部已通过 configAxios 注入 baseURL 和 token，
 * 并已在响应拦截器中剥离 R<T> 包装（直接返回 data 字段）。
 *
 * 注意：导出/下载任务已迁移至统一任务接口 /api/v1/tasks，
 * 请使用 taskApi.create({ type: 'dataset_export' | 'item_download' | 'batch_download', ... }) 创建任务。
 */
import {
  DatasetAPI,
  DatasetItemAPI,
  ItemFileAPI,
} from 'dehaze-sdk-js';
import type {
  Dataset,
  DatasetItem,
  DatasetImage,
  DatasetQuery,
  DatasetItemQuery,
} from '../types/dataset';
import type { PageResult } from 'dehaze-sdk-js';

/** 分页结果 */
export type DatasetPage = PageResult<Dataset[]>;
export type DatasetItemPage = PageResult<DatasetItem[]>;

export const datasetApi = {
  // ==================== 数据集 ====================

  /** 分页查询数据集列表 */
  fetchDatasets(query?: DatasetQuery): Promise<DatasetPage> {
    return DatasetAPI.getList(query);
  },

  /** 懒加载子数据集 */
  fetchChildren(parentId: number): Promise<Dataset[]> {
    return DatasetAPI.getChildren(parentId);
  },

  /** 获取数据集下拉选项 */
  fetchOptions() {
    return DatasetAPI.getOptions();
  },

  /** 获取数据集详情 */
  fetchDatasetDetail(id: number): Promise<Dataset> {
    return DatasetAPI.getDatasetInfoById(id);
  },

  /** 删除数据集 */
  deleteDataset(id: number) {
    return DatasetAPI.deleteById(id);
  },

  /** 批量删除数据集 */
  batchDeleteDatasets(ids: number[]) {
    return DatasetAPI.batchDelete({ ids });
  },

  // ==================== 数据项 ====================

  /** 分页查询数据项列表 */
  fetchDatasetItems(query?: DatasetItemQuery): Promise<DatasetItemPage> {
    return DatasetItemAPI.getList(query);
  },

  /** 获取数据项详情（含配对图片） */
  fetchItemDetail(id: number): Promise<DatasetItem> {
    return DatasetItemAPI.getById(id);
  },

  /** 删除数据项 */
  deleteItem(id: number) {
    return DatasetItemAPI.deleteById(id);
  },

  /** 批量删除数据项 */
  batchDeleteItems(ids: number[]) {
    return DatasetItemAPI.batchDelete({ ids });
  },

  // ==================== 图片文件 ====================

  /** 获取图片详情 */
  fetchImageDetail(id: number): Promise<DatasetImage> {
    return ItemFileAPI.getById(id);
  },

  /** 删除图片 */
  deleteImage(id: number) {
    return ItemFileAPI.deleteById(id);
  },

  /** 批量删除图片 */
  batchDeleteImages(ids: number[]) {
    return ItemFileAPI.batchDelete({ ids });
  },
};

File: dehaze-android/sdk/src/main/java/com/pei/dehaze/sdk/api/DatasetAPI.java
package com.pei.dehaze.sdk.api;

import com.pei.dehaze.sdk.ApiCallback;
import com.pei.dehaze.sdk.DehazeSDK;
import com.pei.dehaze.sdk.model.Option;
import com.pei.dehaze.sdk.model.PageResult;
import com.pei.dehaze.sdk.model.Result;
import com.pei.dehaze.sdk.model.dataset.BatchDeleteForm;
import com.pei.dehaze.sdk.model.dataset.Dataset;
import com.pei.dehaze.sdk.model.dataset.DatasetImageFileInfo;
import com.pei.dehaze.sdk.model.dataset.DatasetItemCreateForm;
import com.pei.dehaze.sdk.model.dataset.DatasetItemUpdateForm;
import com.pei.dehaze.sdk.model.dataset.DatasetQuery;
import com.pei.dehaze.sdk.model.dataset.ImageItem;
import com.pei.dehaze.sdk.model.dataset.ImageItemQuery;
import com.pei.dehaze.sdk.model.dataset.ImageType;
import com.pei.dehaze.sdk.model.dataset.ItemFileUpdateForm;

import java.io.File;
import java.util.List;

import okhttp3.MediaType;
import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import retrofit2.Call;

/**
 * 数据集相关API接口封装
 */
public class DatasetAPI {

    private DatasetAPI() {
    }

    // ===== 数据集 =====

    /**
     * 分页查询数据集列表
     */
    public static void getList(DatasetQuery query, ApiCallback<PageResult<Dataset>> callback) {
        Call<Result<PageResult<Dataset>>> call = DehazeSDK.getInstance().getDatasetApiService()
                .getDatasetList(query.getPageNum(), query.getPageSize(),
                        query.getKeyword(), query.getType(), query.getStatus());
        call.enqueue(callback);
    }

    /**
     * 获取完整数据集树
     */
    public static void getTree(ApiCallback<List<Dataset>> callback) {
        Call<Result<List<Dataset>>> call = DehazeSDK.getInstance().getDatasetApiService().getDatasetTree();
        call.enqueue(callback);
    }

    /**
     * 获取数据集下拉选项
     */
    public static void getOptions(ApiCallback<List<Option>> callback) {
        Call<Result<List<Option>>> call = DehazeSDK.getInstance().getDatasetApiService().getDatasetOptions();
        call.enqueue(callback);
    }

    /**
     * 获取子数据集列表（懒加载）
     */
    public static void getChildren(long parentId, ApiCallback<List<Dataset>> callback) {
        Call<Result<List<Dataset>>> call = DehazeSDK.getInstance().getDatasetApiService().getDatasetChildren(parentId);
        call.enqueue(callback);
    }

    /**
     * 根据ID获取数据集详情
     */
    public static void getDatasetInfoById(long id, ApiCallback<Dataset> callback) {
        Call<Result<Dataset>> call = DehazeSDK.getInstance().getDatasetApiService().getDatasetById(id);
        call.enqueue(callback);
    }

    /**
     * 新增数据集
     */
    public static void add(Dataset data, ApiCallback<Void> callback) {
        Call<Result<Void>> call = DehazeSDK.getInstance().getDatasetApiService().addDataset(data);
        call.enqueue(callback);
    }

    /**
     * 修改数据集
     */
    public static void update(long id, Dataset data, ApiCallback<Void> callback) {
        Call<Result<Void>> call = DehazeSDK.getInstance().getDatasetApiService().updateDataset(id, data);
        call.enqueue(callback);
    }

    /**
     * 删除单个数据集
     */
    public static void delete(long id, ApiCallback<Void> callback) {
        Call<Result<Void>> call = DehazeSDK.getInstance().getDatasetApiService().deleteDataset(id);
        call.enqueue(callback);
    }

    /**
     * 批量删除数据集
     */
    public static void batchDelete(BatchDeleteForm form, ApiCallback<Void> callback) {
        Call<Result<Void>> call = DehazeSDK.getInstance().getDatasetApiService().batchDeleteDatasets(form);
        call.enqueue(callback);
    }

    // ===== 数据项 =====

    /**
     * 分页查询数据项列表
     */
    public static void getItems(ImageItemQuery query, ApiCallback<PageResult<ImageItem>> callback) {
        Call<Result<PageResult<ImageItem>>> call = DehazeSDK.getInstance().getDatasetApiService()
                .getDatasetItems(query.getDatasetId(), query.getPageNum(), query.getPageSize(),
                        query.getKeyword(), query.getSceneType(), query.getHazeLevel());
        call.enqueue(callback);
    }

    /**
     * 获取数据项详情
     */
    public static void getItemById(long itemId, ApiCallback<ImageItem> callback) {
        Call<Result<ImageItem>> call = DehazeSDK.getInstance().getDatasetApiService().getDatasetItemById(itemId);
        call.enqueue(callback);
    }

    /**
     * 创建空数据项
     */
    public static void createItem(DatasetItemCreateForm form, ApiCallback<Long> callback) {
        Call<Result<Long>> call = DehazeSDK.getInstance().getDatasetApiService().createDatasetItem(form);
        call.enqueue(callback);
    }

    /**
     * 修改数据项
     */
    public static void updateItem(long itemId, DatasetItemUpdateForm form, ApiCallback<Void> callback) {
        Call<Result<Void>> call = DehazeSDK.getInstance().getDatasetApiService().updateDatasetItem(itemId, form);
        call.enqueue(callback);
    }

    /**
     * 删除数据项
     */
    public static void deleteItem(long itemId, ApiCallback<Void> callback) {
        Call<Result<Void>> call = DehazeSDK.getInstance().getDatasetApiService().deleteDatasetItem(itemId);
        call.enqueue(callback);
    }

    /**
     * 批量删除数据项
     */
    public static void batchDeleteItems(BatchDeleteForm form, ApiCallback<Void> callback) {
        Call<Result<Void>> call = DehazeSDK.getInstance().getDatasetApiService().batchDeleteDatasetItems(form);
        call.enqueue(callback);
    }

    // ===== 图片文件 =====

    /**
     * 获取图片文件详情
     */
    public static void getItemFileById(long fileId, ApiCallback<DatasetImageFileInfo> callback) {
        Call<Result<DatasetImageFileInfo>> call = DehazeSDK.getInstance().getDatasetApiService().getItemFileById(fileId);
        call.enqueue(callback);
    }

    /**
     * 上传数据项图片
     *
     * @param datasetItemId 数据项ID
     * @param type          图片类型(clear/hazy/trans)
     * @param file          图片文件
     * @param description   描述
     */
    public static void uploadItemFile(long datasetItemId, ImageType type, File file, String description,
                                       ApiCallback<DatasetImageFileInfo> callback) {
        RequestBody requestFile = RequestBody.create(MediaType.parse("image/*"), file);
        MultipartBody.Part filePart = MultipartBody.Part.createFormData("file", file.getName(), requestFile);
        RequestBody itemIdBody = RequestBody.create(MediaType.parse("text/plain"), String.valueOf(datasetItemId));
        RequestBody typeBody = RequestBody.create(MediaType.parse("text/plain"), type.getValue());
        RequestBody descBody = RequestBody.create(MediaType.parse("text/plain"), description != null ? description : "");

        Call<Result<DatasetImageFileInfo>> call = DehazeSDK.getInstance().getDatasetApiService()
                .uploadItemFile(filePart, itemIdBody, typeBody, descBody);
        call.enqueue(callback);
    }

    /**
     * 修改图片信息
     */
    public static void updateItemFile(long fileId, ItemFileUpdateForm form, ApiCallback<Void> callback) {
        Call<Result<Void>> call = DehazeSDK.getInstance().getDatasetApiService().updateItemFile(fileId, form);
        call.enqueue(callback);
    }

    /**
     * 删除图片
     */
    public static void deleteItemFile(long fileId, ApiCallback<Void> callback) {
        Call<Result<Void>> call = DehazeSDK.getInstance().getDatasetApiService().deleteItemFile(fileId);
        call.enqueue(callback);
    }

    /**
     * 批量删除图片
     */
    public static void batchDeleteItemFiles(BatchDeleteForm form, ApiCallback<Void> callback) {
        Call<Result<Void>> call = DehazeSDK.getInstance().getDatasetApiService().batchDeleteItemFiles(form);
        call.enqueue(callback);
    }
}
